### PR TITLE
Add OTel java agent with JMX Metric Insights to kafka

### PR DESCRIPTION
**Pull request by svrnm**
# Changes

This PR adds the java agent to the kafka broker. While this will not provide tracing, it will use [JMX Metric Insights](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/jmx-metrics/javaagent) generate metrics and export them to prometheus:

Screenshot 2022-12-21 at 15 27 02

Screenshot 2022-12-21 at 19 03 26

## Additional info

We plan to have a blog post about JMX Metric Insights on opentelemetry.io eventually, so calling out that the demo is making use of it would be a nice feature:-)

## Merge Requirements

* [x] `CHANGELOG.md` updated to document new feature additions
* [x] Appropriate documentation updates in the [docs](../docs/) folder

### CODE REVIEW

File: src/kafka/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,10 @@
 FROM confluentinc/cp-kafka:latest-ubi8
 
+USER root
```
Comment (fatsheep9146) [inline src/kafka/Dockerfile:3]: Should we allow using the `root` user? I remember that we modify all service user to non-root user in previous pr.

@open-telemetry/demo-approvers @open-telemetry/demo-maintainers

Comment (svrnm) [reply src/kafka/Dockerfile:3]: The root permissions are needed for the chmod of the agent. They are dropped again later down in the Dockerfile.

**Review by mviitane — APPROVED** [review]
Runs fine. Nice to get the additional Kafka metrics. Kafka seems to consume a massive amount of memory, but I guess there's no easy solution for that.

**Review by puckpuck — APPROVED** [review]
This is fantastic. TYVM!

**Pull request merged**